wb/nm_helper/dns_resolver.py

```diff
@@ -0,0 +1,55 @@
+import select
+import socket
+import time
+
+import pycares
+
+
+# Taken from https://github.com/saghul/pycares/blob/master/examples/cares-select.py
+def wait_pycares_channel(channel: pycares.Channel) -> None:
+    start = time.monotonic_ns()
+    while time.monotonic_ns() - start < 5000000000:
+        read_fds, write_fds = channel.getsock()
+        if not read_fds and not write_fds:
+            break
+        timeout = channel.timeout()
+        no_file_d = pycares.ARES_SOCKET_BAD  # pylint: disable=E1101
+        if not timeout:
+            channel.process_fd(no_file_d, no_file_d)
+            continue
+        rlist, wlist, _xlist = select.select(read_fds, write_fds, [], timeout)
+        for file_d in rlist:
+            channel.process_fd(file_d, no_file_d)
+        for file_d in wlist:
+            channel.process_fd(no_file_d, file_d)
+
+
+class DomainNameResolveException(Exception):
+    def __init__(self, msg):
+        self.msg = msg
+
+    def __repr__(self):
+        return self.msg
```

Comment:
```suggestion
class DomainNameResolveException(Exception):
    pass
```

Там конструктор по умолчанию нормально работает